rules/prefer-type-error.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
```

Comment:
I dunno. I've been using `TypeError` for this: https://github.com/sindresorhus/crypto-random-string/blob/21960c996c3f77cf5f7bf6765e837e7c8a07e200/index.js#L5-L7

I guess my thinking is that, even though it's not a different type, it's not the correct form of the type.